Unable to build with ArmNN option enabled

# Issue Description
I am unable to build with ArmNN option enabled. Meson does not find ArmNN.
Upstream now provides cmake files: https://github.com/ARM-software/armnn/commit/eac4adb5bf493ba7d1b00ef6ecd6b56a42ea3f3d

Expected Result
============
meson should find ArmNN cmake files

How to Reproduce
===============
Install ArmNN 20.11 and try to build nnstreamer against it.


Further Information
===============
* Exact OS version: openSUSE Tumbleweed on aarch64

> @ggardet Could you check #3094?

It is still not working, but this time it may be a problem with the content of ArmNN cmake file (`/usr/lib64/libfmt.a` is not present since we only build and package dynamic libs, not static ones):
```
Determining dependency 'armnn' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: :/usr/lib64/pkgconfig:/usr/share/pkgconfig
Called `/usr/bin/pkg-config --modversion armnn` -> 1

CMake binary for MachineChoice.HOST is cached.
None of 'CMAKE_PREFIX_PATH' are defined in the environment, not changing global flags.

Determining dependency 'armnn' with CMake executable '/usr/bin/cmake'
Try CMake generator: auto
Calling CMake (['/usr/bin/cmake']) in /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_armnn with:
  - "-DNAME=armnn"
  - "-DARCHS="
  - "-DCOMPS="
  - "--trace-expand"
  - "--trace-format=json-v1"
  - "--no-warn-unused-cli"
  - "--trace-redirect=cmake_trace.txt"
  - "-DCMAKE_TOOLCHAIN_FILE=/home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_armnn/CMakeMesonToolchainFile.cmake"
  - "."
Run-time dependency armnn found: NO (tried pkgconfig and cmake)
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'Armnn' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: :/usr/lib64/pkgconfig:/usr/share/pkgconfig
Called `/usr/bin/pkg-config --modversion Armnn` -> 1

CMake binary for MachineChoice.HOST is cached.
None of 'CMAKE_PREFIX_PATH' are defined in the environment, not changing global flags.

Determining dependency 'Armnn' with CMake executable '/usr/bin/cmake'
Try CMake generator: auto
Calling CMake (['/usr/bin/cmake']) in /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn with:
  - "-DNAME=Armnn"
  - "-DARCHS="
  - "-DCOMPS="
  - "--trace-expand"
  - "--trace-format=json-v1"
  - "--no-warn-unused-cli"
  - "--trace-redirect=cmake_trace.txt"
  - "-DCMAKE_TOOLCHAIN_FILE=/home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn/CMakeMesonToolchainFile.cmake"
  - "."
CMake failed for generator  and package Armnn with error code 1
OUT:
Running with expanded trace output on.
Not searching for unused variables given on the command line.
Trace will be written to cmake_trace.txt
-- The CXX compiler identification is GNU 10.2.1
-- The C compiler identification is GNU 10.2.1
-- Found Armnn: /usr/lib64/cmake/armnn/ArmnnConfig.cmake
-- Armnn Import: /usr/lib64/cmake/armnn/ArmnnTargets.cmake
-- Configuring incomplete, errors occurred!
See also "/home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn/CMakeFiles/CMakeOutput.log".



ERR:
CMake Error at /usr/lib64/cmake/armnn/ArmnnTargets.cmake:104 (message):
  The imported target "Armnn::fmt" references the file

     "/usr/lib64/libfmt.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/lib64/cmake/armnn/ArmnnTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /usr/lib64/cmake/armnn/ArmnnConfig.cmake:12 (include)
  CMakeLists.txt:15 (find_package)





Try CMake generator: auto
CMake failed for generator  and package Armnn with error code 1
OUT:
Running with expanded trace output on.
Not searching for unused variables given on the command line.
Trace will be written to cmake_trace.txt
-- The CXX compiler identification is GNU 10.2.1
-- The C compiler identification is GNU 10.2.1
-- Found Armnn: /usr/lib64/cmake/armnn/ArmnnConfig.cmake
-- Armnn Import: /usr/lib64/cmake/armnn/ArmnnTargets.cmake
-- Configuring incomplete, errors occurred!
See also "/home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn/CMakeFiles/CMakeOutput.log".



ERR:
CMake Error at /usr/lib64/cmake/armnn/ArmnnTargets.cmake:104 (message):
  The imported target "Armnn::fmt" references the file

     "/usr/lib64/libfmt.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/lib64/cmake/armnn/ArmnnTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /usr/lib64/cmake/armnn/ArmnnConfig.cmake:12 (include)
  CMakeLists.txt:15 (find_package)





Try CMake generator: Ninja
Calling CMake (['/usr/bin/cmake']) in /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn with:
  - "-G"
  - "Ninja"
  - "-DNAME=Armnn"
  - "-DARCHS="
  - "-DCOMPS="
  - "--trace-expand"
  - "--trace-format=json-v1"
  - "--no-warn-unused-cli"
  - "--trace-redirect=cmake_trace.txt"
  - "-DCMAKE_TOOLCHAIN_FILE=/home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn/CMakeMesonToolchainFile.cmake"
  - "."
CMake failed for generator Ninja and package Armnn with error code 1
OUT:
Running with expanded trace output on.
Not searching for unused variables given on the command line.
Trace will be written to cmake_trace.txt
-- The CXX compiler identification is GNU 10.2.1
-- The C compiler identification is GNU 10.2.1
-- Found Armnn: /usr/lib64/cmake/armnn/ArmnnConfig.cmake
-- Armnn Import: /usr/lib64/cmake/armnn/ArmnnTargets.cmake
-- Configuring incomplete, errors occurred!
See also "/home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn/CMakeFiles/CMakeOutput.log".



ERR:
CMake Error at /usr/lib64/cmake/armnn/ArmnnTargets.cmake:104 (message):
  The imported target "Armnn::fmt" references the file

     "/usr/lib64/libfmt.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/lib64/cmake/armnn/ArmnnTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /usr/lib64/cmake/armnn/ArmnnConfig.cmake:12 (include)
  CMakeLists.txt:15 (find_package)





Try CMake generator: Unix Makefiles
Calling CMake (['/usr/bin/cmake']) in /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn with:
  - "-G"
  - "Unix Makefiles"
  - "-DNAME=Armnn"
  - "-DARCHS="
  - "-DCOMPS="
  - "--trace-expand"
  - "--trace-format=json-v1"
  - "--no-warn-unused-cli"
  - "--trace-redirect=cmake_trace.txt"
  - "-DCMAKE_TOOLCHAIN_FILE=/home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn/CMakeMesonToolchainFile.cmake"
  - "."
CMake failed for generator Unix Makefiles and package Armnn with error code 1
OUT:
Running with expanded trace output on.
Not searching for unused variables given on the command line.
Trace will be written to cmake_trace.txt
-- The CXX compiler identification is GNU 10.2.1
-- The C compiler identification is GNU 10.2.1
-- Found Armnn: /usr/lib64/cmake/armnn/ArmnnConfig.cmake
-- Armnn Import: /usr/lib64/cmake/armnn/ArmnnTargets.cmake
-- Configuring incomplete, errors occurred!
See also "/home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn/CMakeFiles/CMakeOutput.log".



ERR:
CMake Error at /usr/lib64/cmake/armnn/ArmnnTargets.cmake:104 (message):
  The imported target "Armnn::fmt" references the file

     "/usr/lib64/libfmt.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/lib64/cmake/armnn/ArmnnTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /usr/lib64/cmake/armnn/ArmnnConfig.cmake:12 (include)
  CMakeLists.txt:15 (find_package)





Try CMake generator: Visual Studio 10 2010
Calling CMake (['/usr/bin/cmake']) in /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn with:
  - "-G"
  - "Visual Studio 10 2010"
  - "-DNAME=Armnn"
  - "-DARCHS="
  - "-DCOMPS="
  - "--trace-expand"
  - "--trace-format=json-v1"
  - "--no-warn-unused-cli"
  - "--trace-redirect=cmake_trace.txt"
  - "-DCMAKE_TOOLCHAIN_FILE=/home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/build/meson-private/cmake_Armnn/CMakeMesonToolchainFile.cmake"
  - "."
CMake failed for generator Visual Studio 10 2010 and package Armnn with error code 1
OUT:



ERR:
CMake Error: Could not create named generator Visual Studio 10 2010

Generators
  Green Hills MULTI            = Generates Green Hills MULTI files
                                 (experimental, work-in-progress).
* Unix Makefiles               = Generates standard UNIX makefiles.
  Ninja                        = Generates build.ninja files.
  Ninja Multi-Config           = Generates build-<Config>.ninja files.
  Watcom WMake                 = Generates Watcom WMake makefiles.
  CodeBlocks - Ninja           = Generates CodeBlocks project files.
  CodeBlocks - Unix Makefiles  = Generates CodeBlocks project files.
  CodeLite - Ninja             = Generates CodeLite project files.
  CodeLite - Unix Makefiles    = Generates CodeLite project files.
  Eclipse CDT4 - Ninja         = Generates Eclipse CDT 4.0 project files.
  Eclipse CDT4 - Unix Makefiles= Generates Eclipse CDT 4.0 project files.
  Kate - Ninja                 = Generates Kate project files.
  Kate - Unix Makefiles        = Generates Kate project files.
  Sublime Text 2 - Ninja       = Generates Sublime Text 2 project files.
  Sublime Text 2 - Unix Makefiles
                               = Generates Sublime Text 2 project files.




Run-time dependency armnn found: NO (tried pkgconfig and cmake)
```

--- a/meson.build
+++ b/meson.build
@@ -320,13 +320,18 @@ foreach feature_name, data : features
   _deps = []
 
   if target != ''
-    target_dep = dependency(target, required: get_option(feature_name))
+    target_dep = dependency(target, required: false)
     if not target_dep.found()
       target_alt = data.get('target_alt', '')
       if target_alt != ''
-        target_dep = dependency(target_alt, required: get_option(feature_name))
+        target_dep = dependency(target_alt, required: false)
       endif
     endif
+
+    if get_option(feature_name).enabled() and not target_dep.found()
+      error('@0@ is enabled but unable to find the target dependency'.format(feature_name))
+    endif
+
     _deps += target_dep
   endif
   _deps += data.get('extra_deps', [])

